File: java/Str.java
class A
{
    public String toString()
    {
        return "[" + "hello world" + "]";
    }
}


public class Str
{
    public static void main(String [] args)
    {
        String str = "hello world";

        System.out.printf("%s\n", str);
        A a = new A();
        System.out.printf("%s\n", a);
        System.out.println(a);
    }
}

File: c++/str.cpp
#include <string>
#include <iostream>

using namespace std;

int main(int argc, char *argv[])
{
    string str = "hello world";

    string::iterator it = str.begin();
    cout << *it << '\n';

    it = str.end();

    cout << *it << '\n';
    
    return 0;
}
